app/scripts/controllers/network/network-controller.test.js

```diff
@@ -85,33 +88,40 @@ async function withController(...args) {
   }
 }
 
+/**
+ * Setup Nock mocks for the block tracker.
+ *
+ * @param {object} options - Options.
+ * @param {NETWORK_TYPES} options.networkType - The network type to mock.
+ * @param {object} options.block - The mock block to return.
+ */
+function setupMockRpcBlockResponses({
+  networkType = NETWORK_TYPES.RPC,
+  block = BLOCK,
+} = {}) {
+  if (networkType === NETWORK_TYPES.RPC) {
+    nock('http://localhost:8545')
+      .persist()
```

Comment:
Hmm, maybe. In some cases it may be both non-deterministic and irrelevant to the purpose of the test, in which case `persist` seems appropriate. But I'm not sure whether that's the case here, I can try it out.

Reply:
It seemed to work out OK! Done here: a54104040733a6c7f1348bf648bc1bdb1a58d31e